### ISSUE TITLE
bug: Nested ion-router-outlet renders as page instead of sub-component

### ISSUE THREAD

**Issue by bcaton85** [ionic-team/ionic#20643]
# Bug Report

**Ionic version:**

[x] **4.x**

**Current behavior:**

Currently using a nested `ion-router-outlet` in one of my components:

```
<h2 id="dashboard-header">Dashboard</h2>
<ion-router-outlet></ion-router-outlet>
```

With the following router configuration:
```
const routes: Routes = [
  { path: '', redirectTo: 'login', pathMatch: 'full' },
  { path: 'login', loadChildren: './login/login.module#LoginModule' },
  { path: 'register', loadChildren: './register/register.module#RegisterModule' },
  { 
    path: 'dashboard', 
    component: Dashboard,
    canActivate: [AuthGuard],
    canActivateChild: [AuthGuard],
    children: [
      { path: 'submit', loadChildren: './submission/submission.module#SubmissionModule' }
    ]
  }
];
```

When hitting the `/dashboard/submit` endpoint the submission component takes the whole screen, appearing over the header. Looking into the styling, I found the following styles for ion-router-outlet:
```
:host {
left: 0;
right: 0;
top: 0;
bottom: 0;
position: absolute;
contain: layout size style;
overflow: hidden;
z-index: 0;
}
```
and the submission component:
```
.ion-page {
left: 0;
right: 0;
top: 0;
bottom: 0;
display: -webkit-box;
display: flex;
position: absolute;
-webkit-box-orient: vertical;
-webkit-box-direction: normal;
flex-direction: column;
-webkit-box-pack: justify;
justify-content: space-between;
contain: layout size style;
overflow: hidden;
z-index: 0;
}
```

The submission component does not have the `IonicPage` annotation. Why is this styling getting applied?

**Expected behavior:**

The `ion-router-outlet` and the `submission` component should not have `absolute` positioning.

**Steps to reproduce:**

-Pull the [following repo](https://github.com/bcaton85/Memories)
-Checkout the `ion-router-issue` tag: `git checkout ion-router-issue`
-`cd frontend`
-`ionic serve`
-Go to /dashboard/submit

The Dashboard header should appear.

**Ionic info:**

Ionic:
   Ionic CLI                     : 5.4.16 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 5.0.0
   @angular-devkit/build-angular : 0.803.25
   @angular-devkit/schematics    : 8.3.25
   @angular/cli                  : 8.3.25
   @ionic/angular-toolkit        : 2.1.2
Capacitor:
   Capacitor CLI   : 1.5.0
   @capacitor/core : 1.5.0
Utility:
   cordova-res : 0.9.0
   native-run  : 0.3.0
System:
   NodeJS : v13.5.0 (/usr/local/Cellar/node/13.5.0/bin/node)
   npm    : 6.13.4
   OS     : macOS Catalina

**Comment by guzmanoj** [comment]
Same issue here.
As a workaround, I'm adding a class to the inner `ion-router-outlet` so that I can  _fix_ the IONIC applied styles.

**Comment by marekbozych** [comment]
+1

**Comment by bcaton85** [comment]
If any one is still having this issue, I was able to resolve it by using the styling provided by ion components, specifically `ion-grid`. Like so:

```
<ion-row>
  <ion-col size="6">
    <ion-item>
      <ion-label>DASHBOAD COMPONENT</ion-label>
    </ion-item>
  </ion-col>
  <ion-col size="6">
    <ion-router-outlet></ion-router-outlet>
  </ion-col>
</ion-row>
```

**Comment by katzenjunge** [comment]
Same issue here.
If you replace `<ion-router-outlet>` with the standard angular `<router-outlet>` everything works as expected and the sub-component does not take up the whole screen anymore.
You lose the fancy transition animation though, so this is only a partial workaround.

**Comment by katzenjunge** [comment]
I was able to fix this issue using the ionic-grid system to arrange my parent-and child component as described here:

https://stackoverflow.com/questions/60443417/ionic-angular-nested-ion-router-outlet-rendering-as-page-instead-of-sub-compon

I still believe this is weird and unexpected behaviour though, expecially since the standard angular `<router-outlet>` works without having to use the grid. At least when using a `component` instead of a `page` the sub-component shouldn't take up the whole screen and exceed the router-outlet that it is nested in imho.

Can anyone clarify?

**Comment by liamdebeasi** [comment]
Thanks for the issue. I am trying out the repo provided, but I am getting the following error: `Error: Cannot match any routes. URL Segment: 'dashboard/submit'`.

I am following these steps:

1. Go to the `frontend` directory.
2. Run `npm i && ionic serve`.
3. Navigate to `/dashboard/submit`.

I see there is a backend server with a Mongo database -- do these need to be used, or does this reproduction work without them?

**Comment by bcaton85** [comment]
Hi Liam. That repo has been progressed past that commit since i posted this. I've added a `ion-router-issue` tag to the commit where i was seeing the issue.

**Comment by guzmanoj** [comment]
Hey Team, 
Any news regarding this?

**Comment by liamdebeasi** [comment]
Apologies for the delay in response. Can you provide some clarification on the steps to reproduce this issue? I am unable to reproduce the issue using the provided repo.

**Comment by guzmanoj** [comment]
> Apologies for the delay in response. Can you provide some clarification on the steps to reproduce this issue? I am unable to reproduce the issue using the provided repo.

Maybe a different scenario, but essentially is the same. This seems to happen only when you use an`ion-router-outlet` within another `ion-router-outlet `(aka nested routes).

- If you wrap the `ion-router-outlet` in an `ion-content` we lose the ability to scroll.
- If you wrap the `ion-router-outlet` in an `ion-content` and also wraps the _inner component_ in another `ion-content` you will get an unwanted _fixed header_ behavior, even though we are not using an `ion-header` (but scroll will be possible)
- If you wrap only the _inner component_ it goes fullscreen and you cannot interact with the other elements in the page 
- Also notice that you can navigate using the `routerLink` only once (I don't know if is related)

As **bcaton85** stated, this seems to be related to the `ion-page` class, which is applied to every direct child of the ion-router-outlet
Please have a look at the stackblitz reproduction
https://stackblitz.com/edit/ionic-4-child-routes-issues

(Manually navigate to https://ionic-4-child-routes-issues.stackblitz.io/cars/123/publish)

**Comment by liamdebeasi** [comment]
@guzmanoj Thanks! I am going to close this as a duplicate of https://github.com/ionic-team/ionic/issues/18197. (It is a different scenario, but the underlying issue appears to be the same).
